lib/source-code/token-store/utils.js

```diff
@@ -30,9 +30,20 @@ function getStartLocation(token) {
  * @returns {number} The found index or `tokens.length`.
  */
 exports.search = function search(tokens, location) {
-    const index = tokens.findIndex(el => location <= getStartLocation(el));
+    for (let minIndex = 0, maxIndex = tokens.length - 1; minIndex <= maxIndex;) {
+        const index = (minIndex + maxIndex) / 2 | 0;
```

Comment:
Could we use `Math.trunc()` here for clarity?

Reply:
I've changed this line to use `Math.trunc()`, but this seems to have a slightly negative impact on performance. Would love if someone else could verify my results independently.

Reply:
I'm getting similar results when comparing several of the slowest rules: binary search with `| 0` (first commit) improves performance by ~10%, but then `Math.trunc()` (second commit) adds back ~5%. Well, since the goal is to improve performance, let's use `| 0`. We can also add a comment that our testing has shown that `| 0` is faster than `Math.trunc()`.